.github/workflows/continuous-integration.yaml

```diff
@@ -0,0 +1,77 @@
+name: Continuous Integration
+
+on:
+  push:
+  schedule:
+    - cron: "5 1 * * *"
+
+jobs:
+  unit-tests:
+    name: Unit Tests
+    runs-on: ubuntu-20.04
+
+    strategy:
+      fail-fast: false
+      matrix:
+        php-versions: ['7.2', '7.3', '7.4']
+        include:
+        - php-versions: 8.0
+          experimental: true
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ matrix.php-versions }}
+          extensions: gd, mysqli
+          #tools: php-cs-fixer, phpunit:${{ matrix.phpunit-versions }}
+
+      - name: Determine composer cache directory
+        id: composer-cache
+        run: echo "::set-output name=directory::$(composer config cache-dir)"
+
+      - name: Cache composer dependencies
+        uses: actions/cache@v1
+        with:
+            path: ${{ steps.composer-cache.outputs.directory }}
+            key: ${{ matrix.php }}-composer-${{ hashFiles('**/composer.lock') }}
+            restore-keys: ${{ matrix.php }}-composer-
+
+      - name: Install dependencies
+        run: composer install
+
+      - name: Run tests
+        run: bin/phpunit
+
+  style-tests:
+    name: Code Style Tests
+    runs-on: ubuntu-20.04
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          # Run style checks only on one version (latest supported)
+          php-version: 7.4
+          extensions: gd, mysqli
+
+      - name: Determine composer cache directory
+        id: composer-cache
+        run: echo "::set-output name=directory::$(composer config cache-dir)"
+
+      - name: Cache composer dependencies
+        uses: actions/cache@v1
+        with:
+            path: ${{ steps.composer-cache.outputs.directory }}
+            key: ${{ matrix.php }}-composer-${{ hashFiles('**/composer.lock') }}
+            restore-keys: ${{ matrix.php }}-composer-
+
+      - name: Install dependencies
+        run: composer install
+
+      - name: Run tests
+        run: bin/phpcs --ignore=*/vendor/* --standard=PSR1,PSR2 --exclude=Generic.Files.LineLength,PSR1.Methods.CamelCapsMethodName --extensions=php -s .
```

Comment:
This appears to be a functional regression (and also causing the check to fail)
Instead of checking modified files, it appears that syntax check is applied to all files, which leads to longer execution time and also the problem that syntax has to be fixed everywhere for this to pass

Reply:
modified part to only call PHPCS on changed files